File: androidApp/features/profile/src/main/kotlin/com/velkonost/getbetter/android/features/profile/components/OtherBlock.kt
package com.velkonost.getbetter.android.features.profile.components

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.padding
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.velkonost.getbetter.core.compose.components.PrimaryBox
import com.velkonost.getbetter.shared.resources.SharedR
import dev.icerock.moko.resources.compose.painterResource
import dev.icerock.moko.resources.compose.stringResource

@Composable
fun OtherBlock(
    modifier: Modifier = Modifier,
    onPrivacyClick: () -> Unit,
    onTermsClick: () -> Unit
) {

    SectionTitle(text = stringResource(resource = SharedR.strings.profile_help_title))

    PrimaryBox {
        Column {
            MenuItem(
                title = stringResource(resource = SharedR.strings.profile_privacy_title),
                icon = painterResource(imageResource = SharedR.images.ic_privacy),
                onClick = onPrivacyClick
            )

            MenuItem(
                modifier = modifier.padding(top = 4.dp),
                title = stringResource(resource = SharedR.strings.profile_terms_title),
                icon = painterResource(imageResource = SharedR.images.ic_terms),
                onClick = onTermsClick
            )
        }
    }

}

File: iosApp/iosApp/Features/Profile/components/OtherBlock.swift
//
//  OtherBlock.swift
//  iosApp
//
//  Created by velkonost on 11.01.2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import SwiftUI
import SharedSDK

struct OtherBlock: View {
    
    let onPrivacyClick: () -> Void
    let onTermsClick: () -> Void
    
    init(onPrivacyClick: @escaping () -> Void, onTermsClick: @escaping () -> Void) {
        self.onPrivacyClick = onPrivacyClick
        self.onTermsClick = onTermsClick
    }
    
    var body: some View {
        SectionTitle(text: SharedR.strings().profile_other_title.desc().localized())
        
        PrimaryBox {
            VStack {
                MenuItem(
                    title: SharedR.strings().profile_privacy_title.desc().localized(),
                    icon: SharedR.images().ic_privacy.toUIImage()!,
                    onClick: onPrivacyClick                )
                
                MenuItem(
                    title: SharedR.strings().profile_terms_title.desc().localized(),
                    icon: SharedR.images().ic_terms.toUIImage()!,
                    onClick: onTermsClick
                )
            }
        }
    }
    
}
